### PR TITLE
Chaining click events (double/triple click etc)

### PR DESCRIPTION
**Pull request by darrenburns**
- Adds support for detecting double/triple clicks via `chain` attribute on `Click` events.
- Updates `pilot.click` with a `times` argument, allowing double/triple/etc clicks.

This PR doesn't implement any features which make use of double/triple clicks.

**Please review the following checklist.**

- [x] Docstrings on all new or modified functions / classes 
- [x] Updated documentation
- [x] Updated CHANGELOG.md (where appropriate)

### CODE REVIEW

File: src/textual/app.py [hunk 1]
```diff
@@ -4325,10 +4355,12 @@ def suspend(self) -> Iterator[None]:
             # app, and we don't want to have the driver auto-restart
             # application mode when the application comes back to the
             # foreground, in this context.
+            # fmt: off
```
Comment (darrenburns) [inline src/textual/app.py:4358]: Ruff formats this code into parentheses, and you can't use parentheses in context managers on 3.8.

Comment (darrenburns) [reply src/textual/app.py]: Fixed by adding:

```toml
[tool.ruff]
target-version = "py38"
```

To `pyproject.toml`.

**Review by willmcgugan — APPROVED** [review]
Some comments for your consideration.

File: docs/guide/testing.md [hunk 2]
```diff
@@ -138,6 +138,15 @@ Here's how you would click the line *above* a button.
 await pilot.click(Button, offset=(0, -1))
 ```
 
+### Double & triple clicks
+
+You can simulate double and triple clicks by setting the `times` parameter.
+
+```python
+await pilot.click(Button, times=2)  # Double click
```
[the remark this answers is not in the document]
Comment (willmcgugan) [reply docs/guide/testing.md:146]: How about a `pilot.double_click` and `pilot.triple_click` (in addition to the `times` parameter?

[the remark this answers is not in the document]
Comment (darrenburns) [reply docs/guide/testing.md:146]: Added these

**Pull request merged**